src/pages/ReportDetailsPage.tsx

```diff
@@ -341,13 +367,7 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
                             />
                         </OfflineWithFeedback>
                     )}
-                    {isGroupChat && (
-                        <PromotedActionsBar
-                            report={report}
-                            promotedActions={[PromotedActions.pin(report)]}
-                            shouldShowLeaveButton
-                        />
-                    )}
+                    <PromotedActionsBar promotedActions={[PromotedActions.pin(report), ...(isGroupDMChat ? [] : [PromotedActions.share(report)])]} />
```

Comment:
NAB but I think it is slightly over engineered to have "pin" and "share" functions that return an object. No need for all these different methods IMO. Just use the objects.